src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double DEFAULT_LOAD = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] innerArray;
+    private int size;
+
+    public MyHashMap() {
+        innerArray = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        grow();
+        int index = getIndex(key);
+        Node<K, V> tempNode = innerArray[index];
+        if (tempNode == null) {
+            innerArray[index] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        Node<K, V> prevNode = tempNode;
+        while (tempNode != null) {
+            if (Objects.equals(tempNode.key, key)) {
+                tempNode.value = value;
+                return;
+            }
+            prevNode = tempNode;
+            tempNode = tempNode.next;
+        }
+        prevNode.next = new Node<>(key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> tempNode = innerArray[getIndex(key)];
+        if (tempNode == null) {
+            return null;
+        }
+        if (Objects.equals(tempNode.key, key)) {
+            return tempNode.value;
+        }
```

Comment:
this checks are useless. if `tempNode` is `null`, program return `null`. If `tempNode.key` is equal to `key`, program return `temp.velue`.
```suggestion
```